Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -940,7 +940,7 @@ class ConstantFoldingPhase : public Phase {
                 else
                     radix = m_state.forNode(node->child2()).m_value;
 
-                if (!radix.isNumber())
+                if (!radix.isInt32())
                     break;
 
                 if (radix.asNumber() == 0 || radix.asNumber() == 10) {
```

Comment:
It's a little weird the code below this is removing the type check (`node->child2() = Edge()`). Viewed in another way, it's also contributed to this bug. I wonder why we have that LOC.